tests/smoke.rs

```diff
@@ -887,3 +908,44 @@ tag = "1.0.25"
         run!()
     }
 }
+
+#[cfg(feature = "dev")]
+#[test]
+fn cargo_check_plain() {
+    fn run() -> Result<()> {
+        let project = HProject::new(false)?;
+        project.xargo_check_subcommand(None)?;
+
+        Ok(())
+    }
+    run!()
+}
+
+#[cfg(feature = "dev")]
+#[test]
+fn cargo_check_check() {
+    fn run() -> Result<()> {
+        let project = HProject::new(false)?;
+        project.xargo_check_subcommand(Some("check"))?;
+
+        Ok(())
+    }
+    run!()
+}
+
+#[cfg(feature = "dev")]
+#[test]
+fn cargo_check_check_no_ctoml() {
+    fn run() -> Result<()> {
+        let project = HProject::new(false)?;
+        project.xargo_toml("")?;
```

Comment:
So I guess `HProject` does not per default create a `Xargo.toml`? If so, please add a comment here like "make sure Xargo.toml exists" or so.